addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js

```diff
@@ -487,45 +406,66 @@ export class FloorScreen extends Component {
                 selectedTables.forEach((selectedTable) => {
                     if (newSeatsNum !== selectedTable.seats) {
                         selectedTable.seats = newSeatsNum;
-                        this._save(selectedTable);
+                        this.pos.updateTables(selectedTable);
                     }
                 });
             },
         });
     }
-    async changeToCircle() {
-        await this.changeShape("round");
+    stopOrderTransfer() {
+        this.pos.isTableToMerge = false;
+        this.pos.orderToTransfer = null;
     }
-    async changeToSquare() {
-        await this.changeShape("square");
+    changeShape(form) {
+        for (const table of this.selectedTables) {
+            table.shape = form;
+        }
+        this.pos.updateTables(...this.selectedTables);
     }
-    async changeShape(form) {
-        if (this.selectedTables.length == 0) {
-            return;
+    unlinkTables() {
+        for (const table of this.selectedTables) {
+            table.update({ parent_id: null });
         }
-        this.selectedTables.forEach(async (selectedTable) => {
-            selectedTable.shape = form;
-            await this._save(selectedTable);
-        });
+        this.pos.updateTables(...this.selectedTables);
     }
-    async setTableColor(color) {
-        const selectedTables = this.selectedTables;
-        selectedTables.forEach(async (selectedTable) => {
-            selectedTable.color = color;
-            await this._save(selectedTable);
-        });
+    setColor(color) {
+        if (this.selectedTables.length > 0) {
+            this.selectedTables.forEach((selectedTable) => {
+                selectedTable.color = color;
+            });
+            this.pos.updateTables(...this.selectedTables);
+        } else {
+            this.activeFloor.background_color = color;
+            this.pos.data.write("restaurant.floor", [this.activeFloor.id], {
+                background_color: color,
+            });
```

Comment:
You don't need to use write from the models when isn't a related records (many2one, one2many etc..) you can directly assign the new value

```suggestion
           this.activeFloor.background_color = color,
```